tools/cli/adminFailoverCommands.go

```diff
@@ -0,0 +1,275 @@
+	if err != nil {
+		ErrorAndExit("Failed to start failover workflow", err)
+	}
+	fmt.Println("Failover workflow started")
+	fmt.Println("wid: " + wf.ID)
+	fmt.Println("rid: " + wf.RunID)
+}
+
+func validateStartParams(params *startParams) {
+	if len(params.targetCluster) == 0 {
+		ErrorAndExit("targetCluster is not provided", nil)
+	}
+	if len(params.sourceCluster) == 0 {
+		ErrorAndExit("sourceCluster is not provided", nil)
+	}
+	if params.targetCluster == params.sourceCluster {
+		ErrorAndExit("targetCluster is same as sourceCluster", nil)
+	}
+	if params.batchFailoverSize <= 0 {
+		params.batchFailoverSize = defaultBatchFailoverSize
+	}
+	if params.batchFailoverWaitTimeInSeconds <= 0 {
+		params.batchFailoverWaitTimeInSeconds = defaultBatchFailoverWaitTimeInSeconds
+	}
+	if params.failoverTimeout <= 0 {
+		params.failoverTimeout = defaultFailoverTimeoutInSeconds
+	}
+}
+
+// AdminFailoverStart start failover workflow
+func AdminFailoverStart(c *cli.Context) {
+	params := &startParams{
+		targetCluster:                  getRequiredOption(c, FlagTargetCluster),
+		sourceCluster:                  getRequiredOption(c, FlagSourceCluster),
+		batchFailoverSize:              c.Int(FlagFailoverBatchSize),
+		batchFailoverWaitTimeInSeconds: c.Int(FlagFailoverWaitTime),
+		failoverTimeout:                c.Int(FlagFailoverTimeout),
+		domains:                        c.StringSlice(FlagFailoverDomains),
+	}
+	failoverStart(c, params)
+}
+
+// AdminFailoverPause pause failover workflow
+func AdminFailoverPause(c *cli.Context) {
+	err := executePauseOrResume(c, true)
+	if err != nil {
+		ErrorAndExit("Failed to pause failover workflow", err)
+	}
+	fmt.Println("Failover paused")
+}
+
+// AdminFailoverResume resume a paused failover workflow
+func AdminFailoverResume(c *cli.Context) {
+	err := executePauseOrResume(c, false)
+	if err != nil {
+		ErrorAndExit("Failed to resume failover workflow", err)
+	}
+	fmt.Println("Failover resumed")
+}
+
+func executePauseOrResume(c *cli.Context, isPause bool) error {
+	client := getCadenceClient(c)
+	tcCtx, cancel := newContext(c)
+	defer cancel()
+
+	runID := getRunID(c, client)
+	var signalName string
+	if isPause {
+		signalName = failoverManager.PauseSignal
+	} else {
+		signalName = failoverManager.ResumeSignal
+	}
+
+	return client.SignalWorkflow(tcCtx, failoverManager.WorkflowID, runID, signalName, nil)
+}
+
+// AdminFailoverQuery query a failover workflow
+func AdminFailoverQuery(c *cli.Context) {
+	client := getCadenceClient(c)
+	tcCtx, cancel := newContext(c)
+	defer cancel()
+
+	var runID string
+	var descResp *shared.DescribeWorkflowExecutionResponse
+	var err error
+	if c.IsSet(FlagRunID) {
+		runID = c.String(FlagRunID)
+	} else {
+		descResp, err = client.DescribeWorkflowExecution(tcCtx, failoverManager.WorkflowID, "")
+		if err != nil {
+			ErrorAndExit("Failed to get failover workflow runID", err)
+		}
+		runID = descResp.WorkflowExecutionInfo.Execution.GetRunId()
+	}
```

Comment:
If empty runID is used for `QueryWorkflow`, it will automatically use the current runID I believe.

Reply:
I improved this part